changelogs/unreleased/4621-skriss-minor.md

```diff
@@ -0,0 +1,4 @@
+## Update to Envoy 1.23.0
+
+Contour now uses Envoy 1.23.0.
+See [the Envoy changelog](https://www.envoyproxy.io/docs/envoy/v1.23.0/version_history/v1.23/v1.23.0) for more information on the contents of the release.
```

Comment:
one thing to note here that may be interesting for operators is to take note of some of the stats name changes

from the release notes:
```
stats listener: fixed metric tag extraction so that [stat_prefix](https://www.envoyproxy.io/docs/envoy/v1.23.0/api-v3/config/listener/v3/listener.proto#envoy-v3-api-field-config-listener-v3-listener-stat-prefix) is properly extracted. This changes the Prometheus name from envoy_listener_myprefix_downstream_cx_overflow{} to envoy_listener_downstream_cx_overflow{envoy_listener_address="myprefix"}. This does not affect the Prometheus name if stat_prefix is not set.
```

Reply:
Seems like this would only affect listeners that specify a `stat_prefix` (https://github.com/envoyproxy/envoy/issues/16103), which ours don't as far as I see, they're getting the default `address_port`

Reply:
ah yeah good call, i'm confusing the stat prefix from other places!